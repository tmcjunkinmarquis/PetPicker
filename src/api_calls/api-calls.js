const fetchUsers = ()=>{
  const url = 'locslhost: 3000 / api / v1 / users'

  console.log('I will fetch users')
}

const fetchPets = ()=>{
  const url = 'https://pet-picker-api.herokuapp.com/api/v1/users/1/pets'
  
  return fetch(url)
  .then((response)=>{
    return response.json()
  })
  .then((pets)=>{
    return pets
  })
  
}

export { fetchUsers, fetchPets };
