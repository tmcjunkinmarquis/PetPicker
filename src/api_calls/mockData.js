export const mockUserData = {
  id: 2, 
  name: "Theresa", 
  species_to_adopt: "dog", 
  search_radius: 5, lat: 0.333e2, 
  lng: 0.444e2, 
  role: "adopter", 
  created_at: "2018-09-18 00:43:08", 
  updated_at: "2018-09-18 00:43:08", 
  password_digest: "$2a$10$5Zm81iRt83CJ.Z4cp79BL.esDeYk8mLpsTzc3BhOA72...", 
  pic: "https://avatars2.githubusercontent.com/u/25600007?...", 
  description: "I absolutely love pups!"
};

export const mockPetData = {
  "id": 1,
  "name": "Bubba",
  "description": "I am a cute pupper",
  "pic": "http://www.bubba.com"
};

export const mockWelcomePetData = {
  id: 1, 
  name: "Wanda", 
  species: 0, 
  description: "Doggo ipsum heckin angery woofer wow such tempt fl...", 
  pic: "https://www.cesarsway.com/sites/newcesarsway/files...",  
  user_id: 4
};

export const mockMatchData = [
  {
    "id": 51,
    "name": "Mack",
    "description": "Poops a lot",
    "pic": "https://cdn2-www.dogtime.com/assets/uploads/2011/03/puppy-development-460x306.jpg"
  },
  {
    "id": 52, 
    "name":  "Felton", 
    "description": "Poops a lot", 
    "pic": "https://cdn2-www.dogtime.com/assets/uploads/2011/03/puppy-development-460x306.jpg"
  },
  { "id": 53, 
    "name": "Tana", 
    "description": "Poops a lot", 
    "pic": "https://cdn2-www.dogtime.com/assets/uploads/2011/03/puppy-development-460x306.jpg" }
];

