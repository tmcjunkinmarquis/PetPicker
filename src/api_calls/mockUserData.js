export const mockUserData = {
  "id": 1,
  "username": "Bubba",
  "password": "123",
  "description": "I am Bubba."
};