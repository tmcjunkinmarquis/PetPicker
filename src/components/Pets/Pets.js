import React, { Component } from 'react';
import { connect } from 'react-redux';
import NopeButton from '../NopeButton/NopeButton';
import LikeButton from '../LikeButton/LikeButton';
import { fetchPets } from '../../api_calls/api-calls';
import PetDescription from '../PetDescription/PetDescription';
import './Pets.css';
import PropTypes from 'prop-types';
import * as actions from '../../actions';

export class Pets extends Component {
  constructor(props) {
    super(props);
    this.state = {
    };
  }

  loadAllPets = async () => {
    if (this.props.loggedIn) {
      const petsArray = await fetchPets(this.props.userId);

      if (!petsArray.length) {
        alert('There are no more pets');
        return;
      }
      this.props.storeAllPets(petsArray);
    } else {
      this.props.history.push('/login');
    }
  }

  loadPet = () => {
    if (this.props.petsArray.length) {
      return (
        <div>
        <h3>{this.props.petsArray[0].name}</h3>
        <img className="responsive-img pets-img"
        src={this.props.petsArray[0].pic}
        alt="pic"
        />
        <div className="nope-or-like">
          <NopeButton />
          <LikeButton />
        </div>
        <p className="flow-text">{this.props.petsArray[0].description}</p>
        </div>
      );
    } else {
      this.loadAllPets();
    }
  }


  componentDidMount = async () => {
    await this.loadAllPets();
  }


  render() {
    return (
      <div>
        {this.loadPet()}
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  loggedIn: state.loggedIn,
  userId: state.user.id,
  petsArray: state.pets.petsArray
});

export const mapDispatchToProps = dispatch => ({
  storeAllPets: petsArray => dispatch(actions.petsArray(petsArray))
});

Pets.propTypes = {
  loggedIn: PropTypes.bool.isRequired,
  userId: PropTypes.number,
  history: PropTypes.object.isRequired,
  petsArray: PropTypes.array,
  storeAllPets: PropTypes.func
};

export default connect(mapStateToProps, mapDispatchToProps)(Pets);
