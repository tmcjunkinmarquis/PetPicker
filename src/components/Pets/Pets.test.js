/* eslint-disable max-len */

import React from 'react';
import { shallow } from 'enzyme';
import { Pets } from './Pets';
import { fetchPets } from '../../api_calls/api-calls';
jest.mock('../../api_calls/api-calls.js');

describe('Pets', () => {
  let wrapper;
  let mockProps;

  beforeEach(() => {
    jest.resetAllMocks();
    mockProps = {
      loggedIn: false,
      userId: 1,
      petsArray: [{}, {}],
      storeAllPets: jest.fn(),
      history: { push: jest.fn() }
    };
    wrapper = shallow(<Pets {...mockProps} />, {
      disableLifecycleMethods: true
    });
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('loadAllPets', async () => {
    it('calls fetchPets if loggedIn', () => {
      const petsArray = [{ name: 'Fido' }];
      mockProps.loggedIn = true;
      wrapper = shallow(<Pets {...mockProps} />, {
        disableLifecycleMethods: true
      });
      fetchPets.mockImplementation(() => Promise.resolve(petsArray));
      expect();
    });
  });
});
