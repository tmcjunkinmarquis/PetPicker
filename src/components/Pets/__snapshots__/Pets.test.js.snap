// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pets matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Pets
    history={
        Object {
            "push": [Function],
          }
    }
    loggedIn={false}
    petsArray={
        Array [
            Object {},
            Object {},
          ]
    }
    storeAllPets={[Function]}
    userId={1}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <h3 />
        <img
                alt="pic"
                className="responsive-img pets-img"
                src={undefined}
        />
        <div
                className="nope-or-like"
        >
                <Connect(Component) />
                <Connect(Component) />
        </div>
        <p
                className="flow-text"
        />
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3 />,
          <img
            alt="pic"
            className="responsive-img pets-img"
            src={undefined}
/>,
          <div
            className="nope-or-like"
>
            <Connect(Component) />
            <Connect(Component) />
</div>,
          <p
            className="flow-text"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "pic",
            "className": "responsive-img pets-img",
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Connect(Component) />,
              <Connect(Component) />,
            ],
            "className": "nope-or-like",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "flow-text",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <h3 />
          <img
                    alt="pic"
                    className="responsive-img pets-img"
                    src={undefined}
          />
          <div
                    className="nope-or-like"
          >
                    <Connect(Component) />
                    <Connect(Component) />
          </div>
          <p
                    className="flow-text"
          />
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3 />,
            <img
              alt="pic"
              className="responsive-img pets-img"
              src={undefined}
/>,
            <div
              className="nope-or-like"
>
              <Connect(Component) />
              <Connect(Component) />
</div>,
            <p
              className="flow-text"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "pic",
              "className": "responsive-img pets-img",
              "src": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Connect(Component) />,
                <Connect(Component) />,
              ],
              "className": "nope-or-like",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "flow-text",
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
